source/common/router/retry_state_impl.cc

```diff
@@ -225,17 +226,21 @@ RetryStateImpl::parseResetInterval(const Http::ResponseHeaderMap& response_heade
 }
 
 void RetryStateImpl::resetRetry() {
-  if (callback_) {
+  if (backoff_callback_) {
```

Comment:
Should this be "!= nullptr" (or perhaps the one below should just be "if (!next_loop_callback_)" so they're consistent.